crates/ruff_linter/resources/test/fixtures/ruff/RUF027.py

```diff
@@ -0,0 +1,58 @@
+val = 2
+
+def simple_cases():
+    a = 4
+    b = "{a}" # RUF027
+    c = "{a} {b} f'{val}' " # RUF027
+    uppercase = "{a}".to_upper() # RUF027
+
+def escaped_string():
+    a = 4
+    b = "escaped string: {{ brackets surround me }}" # RUF027
+
+def raw_string():
+    a = 4
+    b = r"raw string with formatting: {a}" # RUF027
+
+def print_name(name: str):
+    a = 4
+    print("Hello, {name}!") # RUF027
+    print("The test value we're using today is {a}") # RUF027
+
+def do_nothing(a):
+    return a
+
+def do_nothing_with_kwargs(a, **kwargs):
+    return a, kwargs
+
+def nested_funcs():
+    a = 4
+    print(do_nothing(do_nothing("{a}"))) # RUF027
+    do_nothing_with_kwargs(do_nothing("{a}"), a = 5) # RUF027
+
+
+def alternative_formatter(src, **kwargs):
+    src.format(**kwargs)
+
+# These should not cause an RUF027 message
+def negative_cases():
+    a = 4
+    positive = False
+    """{a}"""
+    "don't format: {a}"
+    c = """  {b} """
+    d = "bad variable: {invalid}"
+    e = "incorrect syntax: {}"
+    json = "{ positive: false }"
+    json2 = "{ 'positive': false }"
+    json3 = "{ 'positive': 'false' }"
+    alternative_formatter("{a}", a = 5)
+    formatted = "{a}".fmt(a = 7)
+    print(do_nothing("{a}".format(a=3)))
```

Comment:
What happens with: `"{a}".format(**kwargs)`?

I'm wondering if we should _always_ ignore strings that are the callee on a `.format` call. It seems like it would only ever be a false positive.

Similarly, what about `"{a}" % foo`, which is the older style of string formatting? We should _probably_ take care to ignore those too. What do you think?

Reply:
> I'm wondering if we should always ignore strings that are the callee on a .format call. It seems like it would only ever be a false positive.

I was considering adding something like that (since that's one of the things `pyanalyze` checks for). Let's do that.

> Similarly, what about "{a}" % foo, which is the older style of string formatting? We should probably take care to ignore those too. What do you think?

We definitely should, good call.

Reply:
The old-style printf formatting doesn't use curly braces anywhere (afaik), so shouldn't need it in this check